### PR TITLE
GFF Handling Test Cases

### PR DESCRIPTION
**Pull request by sgolts**
Testing for genes that span multiple contigs, your function assumes each gene is on a single contig. Also testing for duplicate gene IDs with different positions in the file. The current implementation will overwrite the position of the same gene ID.

### CODE REVIEW

File: tests/test_1_compile_variants_by_gene.py [hunk 1]
```diff
@@ -19,6 +19,60 @@ def test_extract_gene_positions(self):
...
+                'single_gene': {
+                    'contig1': (300, 400)
+                }
+            }
+            self.assertEqual(result, expected_output)
+        finally:
+            if os.path.exists(test_gff_path):
+                os.remove(test_gff_path)
+            
+    def test_duplicate_gene_ids(self):
```
Comment (lyuchengmarvin) [inline tests/test_1_compile_variants_by_gene.py:49]: This is important especially when I want to do gene-trait associations. Need to resolve duplicate gene-ids.

**Comment by lyuchengmarvin** [comment]
I think these are helpful checks to handle variations in gene annotation pipelines and assembly quality from the users.

Thank you so much!
